graphql_compiler/schema_generation/schema_graph.py

```diff
@@ -372,6 +373,34 @@ def get_subclass_sets_from_superclass_sets(superclass_sets):
     return subclass_sets
 
 
+def link_schema_elements(elements, inheritance_sets):
+    """For each edge, link the schema elements it connects to each other."""
+    subclass_sets = get_subclass_sets_from_inheritance_sets(inheritance_sets)
```

Comment:
Doesn't this mean that we'll end up computing subclass sets twice? Once in this function, and once when actually building the SchemaGraph?

This is a bit of a code smell. ( https://en.wikipedia.org/wiki/Code_smell in case you haven't heard the term before)

Reply:
Yes. It does.

Reply:
So another option for this could be passing in the subclass_sets when we construct the SchemaGraph but then we'd have to perform additional validation to check that the subclass_sets and the superclass_sets are consistent.

Reply:
@bojanserafimov thoughts? On one hand, constructing subclass sets twice is somewhat expensive and gives us more places to write buggy code. On the other, having to pass both subclass and superclass sets into the constructor and then needing to validate them against each other makes no sense either. Any ideas on how we can design our way out of this situation?

Reply:
What if we created an inheritance sets named tuple that created the subclass sets from the superclass sets? (It would also be entirely immutable).

Reply:
Yeah, something like that could work. Maybe if the namedtuple contained both the sub- and the superclass sets at once, and was constructed immediately from the toposorted schema information. So instead of calling `_get_superclass_sets_from_schema_data()`, call the function that constructs this namedtuple that contains both sub- and superclass information. Then if we always pass the namedtuple itself instead of its components, we know it was well-formed and don't need to validate.

Reply:
that makes sense

Reply:
Ok, I'll make a separate PR for this.